app.rb

```diff
@@ -0,0 +1,98 @@
+
+module Enumerable
+  def my_each(&prc)
+    length.times do |i|
+      prc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&prc)
+    length.times do |i|
+      prc.call(self[i], i)
+    end
+    self
+  end
+  
+  def my_select(&prc)
+    selected = []
+    my_each do |el|
+      selected << el if prc.call(el) == true
+    end
+    selected
+  end
+  
+  def my_all?(arg = nil)
+    if arg
+      my_each { |ele| return false unless arg === ele }
+    elsif block_given?
+      my_each { |ele| return false unless yield(ele) }
+    else
+      my_each { |ele| return false unless ele }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if arg
+      my_each { |ele| return true unless arg === ele }
+    elsif block_given?
+      my_each { |ele| return true unless yield(ele) }
+    else
+      my_each { |ele| return true unless ele }
+    end
+    false
+  end
```

Comment:
- [ ] `my_any` is not working as the original `any?` method, also is not passing all test cases. 

### `any?` is an Enumerable that has these characteristics:
- Returns `true` if at least one element is not false.
```ruby
[1, false, nil].any?
# true
```
- Returns `false` if all elements are false.
```ruby
[nil, false, nil].any?
# false
```
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending on if the values in the array/range/hash match the condition. the Method is failing when giving a Class, RegEx, and pattern
```ruby
[1, false, nil].any?(Numeric)
# true
```

[Enumerable#any?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-any-3F)